scipy/optimize/_differentialevolution.py

```diff
@@ -469,6 +498,40 @@ def init_population_random(self):
         # reset number of function evaluations counter
         self._nfev = 0
 
+    def init_population_array(self, init):
+        """
+        Initialises the population with a user specified population.
+
+        Parameters
+        ----------
+        init : np.ndarray
+            Array specifying subset of the initial population. The array should
+            have shape (M, len(x)), where len(x) is the number of parameters.
+            The population is clipped to the lower and upper `bounds`.
+        """
+        # make sure you're using a float array
+        popn = np.asfarray(init)
+
+        if (np.size(popn, 0) < 5 or
+                    popn.shape[1] != self.parameter_count or
+                    len(popn.shape) != 2):
+            raise ValueError(self.__init_error_msg)
```

Comment:
Not sure this is a good idea to use the same message here and in `__init__` since these are arguably different errors.

Reply:
Done.